### PR TITLE
Short description added

### LINKED ISSUE #44: Short Description of the plot

### PR DESCRIPTION
**Pull request by elizaan**
### Does this PR close any open issues?
Closes #44 

### Give a longer description of what this PR addresses and why it's needed
The PR adds the short description for upset plots.
**Example 1** 
_Dataset:_ Movie
_Short Description:_ This is an UpSet plot which shows set intersection of 7 sets. All major intersections involve the set Action, and Adventure. The largest intersection is Action, and Adventure, with 104 elements. Other large intersections also involve Action and Romance.
_Short Description in JSON Output:_  'techniqueDescription': 'This is an UpSet plot that visualizes set intersection. To learn about UpSet plots, visit https://upset.app.', 'shortDescription': 'This is an UpSet plot which shows set intersection of 7 sets. All major intersections involve the set Action, and Adventure. The largest intersection is Action, and Adventure, with 104 elements. Other large intersections also involve Action and Romance. ', 'longDescription': '# UpSet Introduction\nThis is an UpSet plot that visualizes set intersection. To learn about UpSet plots, visit https://upset.app.\n\n# Dataset Properties\nThe dataset contains 17 sets and 6303 elements, of which 7 sets are shown in the plot.\n\n# Set Properties\nThe set sizes are diverging a lot, ranging from 68 to 503. The largest set is Action with 503 elements, followed by Romance with 471, Adventure with 283, Children with 251, War with 143, Documentary with 127, and Western with 68.\n\n# Intersection Properties\nThe plot is sorted by size in descending order. There are 32 non-empty intersections, all of which are shown in the plot. The largest 5 intersections are the empty intersection (2395), Just Romance (397), Just Action (299), Just Children (156), and Just Documentary (126).\n\n# Statistical Information\nThe average intersection size is 121, and the median is 7. The 90th percentile is 156, and the 10th percentile is 1. The largest set, Action, is present in 40.6% of all non-empty intersections. The smallest set, Western, is present in 21.9% of all non-empty intersections.\n\n# Trend Analysis\n The intersection sizes peak at a value of 2395 and then drastically flatten down to 1. The empty intersection is the largest by a factor of 6. The empty intersection is present with a size of 2395. An all set intersection is not present. The individual set intersections are large in size. The low degree set intersections lie in medium and small sized intersections. Among the small sized intersections, the high order set intersections are significantly present.\n\n'}

**Example 2**
_Dataset:_ Cancer
_Short Description:_ This is an UpSet plot which shows set intersection of 5 sets. All major intersections involve the set Organismal Injury and Abnormalities, and Cancer. The largest intersection is Cancer, Organismal Injury and Abnormalities, Endocrine Stsytem Disorders, and Reproductive System Disease, with 325 elements. Other large intersections also involve Endocrine Stsytem Disorders, Cancer, and Organismal Injury and Abnormalities. The intersection of all sets is the third largest with 123 elements.
_Short Description in JSON Output:_ {'techniqueDescription': 'This is an UpSet plot that visualizes set intersection. To learn about UpSet plots, visit https://upset.app.', 'shortDescription': 'This is an UpSet plot which shows set intersection of 5 sets. All major intersections involve the set Organismal Injury and Abnormalities, and Cancer. The largest intersection is Cancer, Organismal Injury and Abnormalities, Endocrine Stsytem Disorders, and Reproductive System Disease, with 325 elements. Other large intersections also involve Endocrine Stsytem Disorders, Cancer, and Organismal Injury and Abnormalities. The intersection of all sets is the third largest with 123 elements.', 'longDescription': '# UpSet Introduction\nThis is an UpSet plot that visualizes set intersection. To learn about UpSet plots, visit https://upset.app.\n\n# Dataset Properties\nThe dataset contains 5 sets and 2780 elements, of which 5 sets are shown in the plot.\n\n# Set Properties\nThe set sizes are diverging a lot, ranging from 165 to 743. The largest set is Organismal Injury and Abnormalities with 743 elements, followed by Cancer with 741, Endocrine Stsytem Disorders with 618, Reproductive System Disease with 513, and Cell Cycle with 165.\n\n# Intersection Properties\nThe plot is sorted by size in descending order. There are 10 non-empty intersections, all of which are shown in the plot. The largest 5 intersections are Cancer, Organismal Injury and Abnormalities, Endocrine Stsytem Disorders, and Reproductive System Disease (325), Cancer, Organismal Injury and Abnormalities, and Endocrine Stsytem Disorders (138), Cancer, Organismal Injury and Abnormalities, Endocrine Stsytem Disorders, Reproductive System Disease, and Cell Cycle (123), Cancer, Organismal Injury and Abnormalities, and Reproductive System Disease (60), and Cancer, and Organismal Injury and Abnormalities (55).\n\n# Statistical Information\nThe average intersection size is 74, and the median is 43. The 90th percentile is 325, and the 10th percentile is 2. The largest set, Organismal Injury and Abnormalities, is present in 90.0% of all non-empty intersections. The smallest set, Cell Cycle, is present in 50.0% of all non-empty intersections.\n\n# Trend Analysis\n The intersection sizes peak at a value of 325 and then drastically flatten down to 2. Cancer, Organismal Injury and Abnormalities, Endocrine Stsytem Disorders, and Reproductive System Disease is the largest by a factor of 2. An all set intersection is present with a size of 123. The individual set intersections are small in size. The low degree set intersections lie in large sized intersections. The medium degree set intersections can be seen among large sized intersections. In largest and medium and small sized intersections, the high order set intersections are significantly present.\n\n'}
### Have you added or updated relevant tests?
- [ ] Yes
- [ ] No changes are needed

### Have you added or updated relevant documentation?
- [ ] Yes
- [ ] No changes are needed

### Are there any additional TODOs before this PR is ready to go?

TODOs:
- [ ] ...

### CODE REVIEW

File: src/alttxt/phrases.py [hunk 1]
```diff
@@ -37,8 +37,12 @@
 
     },
     # L3 note: observe which sets are not present in many large intersections
-    "AltText": "This is an UpSet plot which shows set intersection of {{visible_set_count}} sets out of {{set_count}} sets"
-    " and the largest intersection is {{max_intersection_name}} ({{max_intersection_size}}). The plot is sorted by {{sort_type}} and {{visible_non_empty_intersect_count}} non-empty intersections are shown.",
+    "AltText": "This is an UpSet plot which shows set intersection of {{visible_set_count}} sets."
```
Comment (JakeWags) [inline src/alttxt/phrases.py]: This sentence feels grammatically incorrect. "... which shows set intersections", or "which shows the intersections of ... sets." is probably better

File: src/alttxt/phrases.py [hunk 2]
```diff
@@ -37,8 +37,12 @@
...
+    " All major intersections involve the set {{highest_dominent_set}}."
```
Comment (JakeWags) [inline src/alttxt/phrases.py]: variable spelling error:
`highest_dominent_set` should be `highest_dominant_set`

File: src/alttxt/phrases.py (hunk 1 above)
Comment (JakeWags) [reply src/alttxt/phrases.py]: Also, it looked to me like this output "Action, and Adventure", but it should only be one set? This is a set intersection. Maybe I read it wrong.

File: src/alttxt/tokenmap.py [hunk 3]
```diff
@@ -173,7 +173,10 @@ def __init__(
             "list10_dev_outliers": self.dev_outliers(10) if len(self.data.subsets)>=10 else self.dev_outliers(len(self.data.subsets)),
             # 5 largest deviations, listed
             "list5_dev_outliers": self.dev_outliers(5) if len(self.data.subsets)>=5 else self.dev_outliers(len(self.data.subsets)),
-            "category_of_subsets": self.categorize_subsets
+            "category_of_subsets": self.categorize_subsets,
+            "highest_dominent_set": self.find_dominant_sets(),
```
Comment (JakeWags) [inline src/alttxt/tokenmap.py]: spelling (same as above)

File: src/alttxt/tokenmap.py [hunk 4]
```diff
@@ -894,4 +891,93 @@ def calculate_intersection_trend(self) -> str:
...
+        if len(sets) == 2:
+            return f"{sets[0]} and {sets[1]}"
+        elif len(sets) > 2:
+            return ', '.join(sets[:-1]) + ', and ' + sets[-1]
+        else:
+            return sets
+
+    def extract_set_names(self, subset_name):
+        set_names = set(subset_name.split(','))
+        return set_names
```
Comment (JakeWags) [inline src/alttxt/tokenmap.py]: Why not use the setMembership field? This feels like a workaround which shouldn't be needed.

Set Membership per subset should tell you exactly which sets compose the subset
ex: 
```json
"..." {
  "...",
  "setMembership": {
    "setA": "Yes",
    "setB": "No",
    "setC": "Yes",
  }
}

**Review by JakeWags — CHANGES_REQUESTED** [review]
Few comments about spelling, grammar, and one specific comment about the `extract_set_names` function, which should apply to most of the functions relating to getting set membership.

File: src/alttxt/tokenmap.py (hunk 4 above)
Comment (elizaan) [reply src/alttxt/tokenmap.py]: added in this Commit: https://github.com/visdesignlab/upset-alt-txt-gen/pull/45/commits/ce0b2133200d73329f5e595c5a09f08b20f2f28a

**Review by JakeWags — APPROVED** [review]
Looks great. Thanks for changing that.

**Pull request merged**